airflow/executors/kubernetes_executor.py

```diff
@@ -599,13 +599,14 @@ def sync(self) -> None:
                 try:
                     self.kube_scheduler.run_next(task)
                 except ApiException as e:
-                    if e.reason == "BadRequest":
-                        self.log.error("Request was invalid. Failing task")
+                    if e.status in (400, 422):
```

Comment:
Might be nice to bring the comment from the tests here as well to explain why we might see these 2 responses.